test/app/components/content_for_component.rb

```diff
@@ -1,5 +1,12 @@
 # frozen_string_literal: true
 
 class ContentForComponent < ActionView::Component::Base
-  def initialize(*); end
+  validates :content, :title, :footer, presence: true
+
+  attr_reader :content, :title, :footer
+
+  def initialize(title: nil, footer: nil)
```

Comment:
if you're going to set via content_for then things have to be optional in the initializer.